jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -128,7 +128,17 @@ ecma_object_is_constructor (ecma_object_t *obj_p) /**< ecma object */
     return (!ecma_get_object_is_builtin (obj_p) || !ecma_builtin_function_is_routine (obj_p));
   }
 
-  return (type >= ECMA_OBJECT_TYPE_BOUND_FUNCTION);
+  if (type == ECMA_OBJECT_TYPE_BOUND_FUNCTION)
```

Comment:
Isn't this change ES2015 specific?

Reply:
I think that this code could also be executed on previous version, because there is native function **jerry_value_is_constructor**.

I built jerryscript with **tools/build.py --debug --show-opcodes=ON --clean** on **master** (1b01171a) branch and created simple C code:
```c
#include "jerry-core/include/jerryscript.h"

int
main (void)
{
  jerry_init (JERRY_INIT_EMPTY);

  const jerry_char_t script1[] = "Math.sin;";
  const jerry_char_t script2[] = "Math.sin.bind(Math);";

  jerry_value_t parsed_code1 =
      jerry_parse (NULL, 0, script1, sizeof (script1) - 1, JERRY_PARSE_NO_OPTS);
  jerry_value_t parsed_code2 =
      jerry_parse (NULL, 0, script2, sizeof (script2) - 1, JERRY_PARSE_NO_OPTS);

  jerry_value_t script_value1 = jerry_run (parsed_code1);
  jerry_value_t script_value2 = jerry_run (parsed_code2);

  bool is_constructor1 = jerry_value_is_constructor (script_value1);
  bool is_constructor2 = jerry_value_is_constructor (script_value2);

  printf ("%s\n", is_constructor1 ? "true" : "false");
  printf ("%s\n", is_constructor2 ? "true" : "false");

  jerry_release_value (script_value1);
  jerry_release_value (script_value2);
  jerry_release_value (parsed_code1);
  jerry_release_value (parsed_code2);
  jerry_cleanup ();
  return 0;
}
```
Execution of this code show:
```
false
true
```

After this commit, it is:
```
false
false
```

Reply:
I see, I've checked the standards and you are right. Good catch!